components/usage/pkg/apiv1/usage.go

```diff
@@ -408,17 +420,43 @@ func newUsageFromInstance(instance db.WorkspaceInstanceForUsage, pricer *Workspa
 		Kind:                db.WorkspaceInstanceUsageKind,
 		WorkspaceInstanceID: instance.ID,
 		Draft:               draft,
-		Metadata:            nil,
 	}
+
+	startedTime := ""
+	if instance.StartedTime.IsSet() {
+		startedTime = db.TimeToISO8601(instance.StartedTime.Time())
+	}
+	endTime := ""
+	if instance.StoppingTime.IsSet() {
+		endTime = db.TimeToISO8601(instance.StoppingTime.Time())
+	}
+	err := usage.SetMetadataWithWorkspaceInstance(db.WorkspaceInstanceUsageData{
```

Comment:
Due to the serialization err, we need to propagate the error upstream to make it visible. That's the bulk of the changes in this PR